homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -860,7 +795,9 @@ def set_optimistic(attribute, value, color_mode=None, condition_attribute=None):
             and ATTR_RGB_COLOR not in kwargs
             and self._topic[CONF_RGB_COMMAND_TOPIC] is not None
         ):
-            rgb_color = self._rgb_color if self._rgb_color is not None else (255,) * 3
+            rgb_color = (
+                self._attr_rgb_color if self._attr_rgb_color is not None else (255,) * 3
```

Comment:
Is it possible for `_attr_rgb_color` to be falsy in a way that it's still valid for the assignment? If not, I'd suggest
```suggestion
                self._attr_rgb_color or (255, 255, 255)
```